pkg/registry/core/node/strategy.go

```diff
@@ -94,27 +92,12 @@ func (nodeStrategy) PrepareForUpdate(ctx context.Context, obj, old runtime.Objec
 func dropDisabledFields(node *api.Node, oldNode *api.Node) {
 	// Nodes allow *all* fields, including status, to be set on create.
 	// for create
-	if !utilfeature.DefaultFeatureGate.Enabled(features.DynamicKubeletConfig) && oldNode == nil {
+	if oldNode == nil {
 		node.Spec.ConfigSource = nil
 		node.Status.Config = nil
 	}
 
-	// for update
-	if !utilfeature.DefaultFeatureGate.Enabled(features.DynamicKubeletConfig) && !nodeConfigSourceInUse(oldNode) && oldNode != nil {
```

Comment:
I still expected the `if !nodeConfigSourceInUse(oldNode) && oldNode != nil {`

Reply:
I'm a bit rusty on this part, but why wouldn't we just want to clear the field in all cases now that it's deprecated/the feature is disabled?

Reply:
upgrading an API server to 1.24 should not instantly change behavior of 1.23 kubelets using this feature

Reply:
Hmm, that means the node authorizer still needs to allow access to the node configmap too, right?

Reply:
hmm. yeah :-/

I guess that means we drop the capability from nodes and stop allowing new data in in 1.24, and drop the authorizer behavior in 1.26?

Reply:
This PR sets `LockToDefault: true` on the feature gate. Will it still be possible to clear the fields via the API once you upgrade the API server to 1.24? If not, then even if you disable the gate on your kubelet/upgrade kubelet to 1.24 to disable the feature, you have to live with some specified-but-ignored API intent until 1.25?

Reply:
> I guess that means we drop the capability from nodes and stop allowing new data in in 1.24, and drop the authorizer behavior in 1.26?

Yeah I think. But why 1.26 and not 1.25? We marked it deprecated in 1.22, so I think we're in compliance with the 9 month/3 release rule at 1.25? And if no Kubelet will use it in a fully-upgraded 1.24 cluster, should be pretty safe to remove the rest in 1.25?

Reply:
When working with features that span apiserver and kubelet, we have to consider kubelets may be up to 2 versions older than the apiserver. It's reasonable to drop the feature from the kubelet now for 1.24, and in 1.26 we can assume the oldest kubelet in the cluster will be 1.24+ and drop the remainder from the API server.

Reply:
Ah, right!

Reply:
reverted. Just deleted the feature gate check for now

Reply:
If the expectation is that 1.23 Kubelets continue to work/interact with the API the same way even when the API is upgraded to 1.24, do we need to somehow except 1.23 and below Kubelets in these checks too (or just keep allowing the feature to be enabled in the API, with the caveat that it does nothing once your kubelets are on 1.24)?

Reply:
this allows existing Node objects that have the data set to continue to keep it